planner/core/logical_plans.go

```diff
@@ -369,8 +369,8 @@ func (p *LogicalJoin) extractFDForOuterJoin(filtersFromApply []expression.Expres
 // means whether there is a `NullEQ` of a join key.
 func (p *LogicalJoin) GetJoinKeys() (leftKeys, rightKeys []*expression.Column, isNullEQ []bool, hasNullEQ bool) {
 	for _, expr := range p.EqualConditions {
-		leftKeys = append(leftKeys, expr.GetArgs()[0].(*expression.Column))
-		rightKeys = append(rightKeys, expr.GetArgs()[1].(*expression.Column))
+		leftKeys = append(leftKeys, expression.ExtractColumnsFromExpression(expr.GetArgs()[0], nil)...)
```

Comment:
i‘ll change the`buildSemiJoinForSetOperator` in `buildExcept`, add another projection under it like what `updateEQCond` does.

EQ((a-b)，(c-d)), for join keys extracted here, means (a,b), (c,d), the join match means the exact equivalence for a=c, and b = d here， actually, we need a project (a-b) as E，(c-d) as F， and join keys here are simply as(E),(F), the join match means the exact equivalence for E = F here. @winoros